diesel/src/query_builder/insert_statement.rs

```diff
@@ -68,6 +68,18 @@ impl<T, U> AsQuery for InsertStatement<T, U> where
     }
 }
 
+impl<T, U> InsertStatement<T, U> {
+    /// Specify what expression is returned after execution of the `insert`.
+    pub fn returning<E>(self, returns: E) -> InsertQuery<E, InsertStatement<T, U>> where
+        E: Expression,
```

Comment:
This needs to check `SelectableExpression`, as well. Or more likely check that `InsertQuery<E, InsertStatement<T, U>>: Query`, and add the `SelectableExpression` constraint there.